script.js

```diff
@@ -1,5 +1,203 @@
+        console.log("User data: ", userData);
+        currentPres = userData.results;
+        displayPresInformation(currentPres);
+}
+
+function setPresidentCards(president){
+    console.log(president);
+
+    for(const president of presidents){
+        createCard(president);
+    }
+}
+
+function displayCards(presidents){
+    let cardsContainer = document.querySelector('.info-container');
+    clearContainer(cardsContainer);
+
+    let firstPres = presidents[0];
+    let newPresCard = createCard(firstPres);
+    cardsContainer.appendChild(newPresCard);
+}
+
+function createPresNameCards(presidents){
+    console.log("createPresNameCards function called");
+    let PresTitle = document.createElement('p');
+    PresTitle.textContent = `Name: ${president.name}`;
+    PresTitle.classList.add('card-title');
+    PresDetails.appendChild(PresTitle);
+}
+
+
+function filterbyPresidentName(president){
+    console.log("filterbyPresidentName function called");
+    // let PresFilter = document.querySelector('.bottom-container');
+    clearContainer();
+    // createPresNameCards(presidents);
+    let PresTitle = document.createElement('p');
+    PresTitle.textContent = `Name: ${president.name}`;
+    PresDetails.appendChild(PresTitle);
+    for (i = 0; i < 45; i++){
+        console.log(PresData)
+        displayPresInformation(RECpresidentinfo[i]);
+    }
+    return data;
+    
+}
+
+    
+
+function filter(filterValue, presidents){
+    console.log("Filter presidents by: ", filterValue);
+
+    if(filterValue === 'All Information'){
+        setPresidentCards(presidents);
+    }
+    else if (filterValue == 'President Name'){
+        let nameONLY = filterbyPresidentName(presidents);
+        console.log("filter value = president name")
+        console.log("US president names: ", nameONLY )
+        displayCards(presidents);
+    }
+    if (filterValue === "Presidential Number"){
+        filterbyPresidentNumber(presidents);
+    }
+    if(filterValue === "Presidential Term"){
+        filterbyPresidentTerm(presidents);
+    }
+    if(filterValue === "Era"){
+        filterbyPresEra(presidents);
+    }
+
+}
+
+async function fetchJSONData(){
+
+    const information = await fetch(`data/presidents.json`);
+    // console.log("Data: ", information);
+    // console.log("Data status: ", information.body);
+   
+    const data = await information.json();
+    console.log("Data:", data);
+    return data;
+
+}
+
+async function runProgram(){
+    let PresData = await fetchJSONData();
+    console.log(PresData);
+    // fetchJSONData();
+    let RECpresidentinfo = PresData.presidents;
+
+    //for loop (for president in receivedpresidents) to display all presidents
+    for (i = 0; i < 45; i++){
+        console.log(PresData)
+        displayPresInformation(RECpresidentinfo[i]);
+    }
+    
+    let filtersearchbtn = document.querySelector('#fetchsearch-button');
+    filtersearchbtn.addEventListener('click', (e) => SearchBTN(e) );
```

Comment:
You could get rid of this and the whole button so the changes happen immediately with the dropdown